### DISCUSSION TITLE
Crossreference subfigures/subtables

### DISCUSSION THREAD

**Discussion by nithinmkp** [quarto-dev/quarto-cli#8354] [Q&A]
### Description
````{qmd}

```{r}
#| label: tbl-tables
#| tbl-cap: "Tables"
#| tbl-subcap: true
#| layout-ncol: 2

library(knitr)
kable(head(cars))
kable(head(pressure))
```
````

suppose i have a code block like this which produces two tables, i want to refer to second table. But i hav'nt labelled the second one explicitly. How do i refer to this? I read that may be @tbl-table-2 might work. But i want it to produce Table: 1(b) rather than Table 1b.

**Comment by Pierre9344** [comment]
You can do that by using multiple code blocks:
````
see @tbl-tables-cars and @tbl-tables-pressure for subfigures cross references. For the full panel cross reference, use @tbl-tables.


::: {#tbl-tables layout-ncol=2}

```{r}
#| label: tbl-tables-cars
#| tbl-cap: "Tables cars"
#| echo: false
library(knitr)
kable(head(cars))
```

```{r}
#| label: tbl-tables-pressure
#| tbl-cap: "Tables pressure"
#| echo: false

kable(head(pressure))
```

Famous Tables
:::
````
It will render as: 


![Capture d'écran 2024-01-19 090602](https://github.com/quarto-dev/quarto-cli/assets/86181677/e5d82a30-7c0f-4c4f-96df-ca28e8f38fd5)

**Comment by nithinmkp** [comment]
Thanks for chipping in . So we cannot do this without using div
environments? and the rendering is still Table 1a. What I want is Table:
1(a). Is there a way??

Thanks


[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
01/19/24,
04:31:11 PM

**Reply by mcanouil** [reply]
I am not sure what you mean.

A cross-reference object is an object with an ID using a prefix.
To be able to cross-reference the sub-components of your "big" table, the sub-tables needs to have labels/ids.
Sub figures/tables don't get IDs, they get position number appended to the main label/id.

See:
- https://quarto.org/docs/authoring/cross-reference-options.html#references
- https://quarto.org/docs/authoring/cross-references-custom.html
- https://quarto.org/docs/authoring/cross-references-divs.html

And somehow related:
- https://github.com/quarto-dev/quarto-cli/issues/3498

**Comment by nithinmkp** [comment]
let me show you an example

````{qmd}
```{python}
#| label: fig-plots
#| fig-cap: "Plots" 
#| fig-subcap:
#|   - "Plot 1"
#|   - "Plot 2" 
#| layout-ncol: 2

import matplotlib.pyplot as plt
plt.plot([1,23,2,4])
plt.show()

plt.plot([8,65,23,90])
plt.show()
```

See @fig-plots for examples. In particular, @fig-plots-2.

````

In quarto website it also mentions "Note that subfigure reference labels are created automatically based on the main chunk label (e.g. @fig-plots-1, @fig-plots-2, etc.)."

Now what i want is reference to subfigure be like this "Figure 1(b)" and not like "Figure 1b". Hope it is clear

**Reply by mcanouil** [reply]
> suppose i have a code block like this which produces two tables, i want to refer to second table. But i hav'nt labelled the second one explicitly. How do i refer to this? I read that may be @tbl-table-2 might work. But i want it to produce Table: 1(b) rather than Table 1b.

> In quarto website it also mentions "Note that subfigure reference labels are created automatically based on the main chunk label (e.g. @fig-plots-1, @fig-plots-2, etc.)."
>
> Now what i want is reference to subfigure be like this "Figure 1(b)" and not like "Figure 1b". Hope it is clear

Did you try? Did you look at the links I shared to customise cross-reference style and type?
At some point, you need to try the code from the documentation and shared by Pierre and me.

Your example is not reproducible as you don't show the format.

<table>
<tr><th>Quarto document</th><th>HTML</th></tr>
<tr>
<td>

````qmd
---
title: "Quarto Playground"
format: html
---

```{r}
#| label: tbl-tables
#| tbl-cap: "Tables"
#| tbl-subcap:
#|   - cars
#|   - pressure
#| layout-ncol: 2

library(knitr)
kable(head(cars))
kable(head(pressure))
```

See @tbl-tables for examples. In particular, @tbl-tables-2.

```{r}
#| label: fig-figures
#| fig-cap: "Figure"
#| fig-subcap:
#|   - Plot 1
#|   - Plot2
#| layout-ncol: 2

plot(1)
plot(2)
```

See @fig-figures for examples. In particular, @fig-figures-2.

````

</td>
<td>
<img width="420" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/9f4900a9-34ee-4e5e-8ab8-6f93b104b527">
</td>
</tr>
</table>

**Reply by nithinmkp** [reply]
Yes, I did. sorry for not providing a reproducible example. Here it is

````{quarto}
---
title: "Untitled"
format: pdf

---

Hello

```{r}
#| label: fig-figures
#| fig-cap: "Figure"
#| fig-subcap:
#|   - Plot 1
#|   - Plot2
#| layout-ncol: 2

plot(1)
plot(2)
```

See @fig-figures for examples. In particular, @fig-figures-2.

```{r}
#| label: tbl-tables
#| tbl-cap: "Tables"
#| tbl-subcap:
#|   - cars
#|   - pressure
#| layout-ncol: 2

library(knitr)
kable(head(cars))
kable(head(pressure))
```

See @tbl-tables for examples. In particular, @tbl-tables-2.

````

**Reply by nithinmkp** [reply]
I think the issue is in PDF and word files.

**Reply by mcanouil** [reply]
Word theming is controlled by Word style, see <https://quarto.org/docs/output-formats/ms-word-templates.html>.
PDF is LaTeX, you can use `keep-tex: true` and play around the code to change via `include-*` or using partials as described in the documentation.

**Comment by nithinmkp** [comment]
Tried to do something like this

````{latex}
% preamble.tex
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{subcaption}
\usepackage{cleveref}

% Customize cref format for main figure
\Crefformat{figure}{#2Figure~#1#3}

% Customize cref format for subfigure
\Crefformat{subfigure}{#2Figure~#1(#3)#4}
````

````{qmd}
---
title: "Untitled"
format:
  pdf:
    documentclass: article
    keep-tex: true
    include-in-header:
      - file: preamble.tex


---

Hello

```{r}
#| label: fig-figures
#| fig-cap: "Figure"
#| fig-subcap:
#|   - Plot 1
#|   - Plot2
#| layout-ncol: 2

plot(1)
plot(2)
```

See @fig-figures for examples. In particular, \Cref{subfig-figures-2}.

```{r}
#| label: tbl-tables
#| tbl-cap: "Tables"
#| tbl-subcap:
#|   - cars
#|   - pressure
#| layout-ncol: 2

library(knitr)
kable(head(cars))
kable(head(pressure))
```

See @tbl-tables for examples. In particular, @tbl-tables-2.
````

Still not getting desired output.
